telethon/tl/custom/conversation.py

```diff
@@ -312,9 +313,13 @@ async def _check_custom(self, built):
         for key, (ev, fut) in list(self._custom.items()):
             ev_type = type(ev)
             inst = built[ev_type]
-            if inst and ev.filter(inst):
-                fut.set_result(inst)
-                del self._custom[key]
+
+            if inst:
+                filter = ev.filter(inst)
+                filter = (await filter) if inspect.isawaitable(filter) else filter
```

Comment:
It's ugly we need to duplicate this but I guess there's no other way around really.

Reply:
Yes, If you don't want to jump into event loop I don't know other way :/

Reply:
Also: should I replace in-line if with full like in second place?

Reply:
Yes, better to write a proper `if` block.